### ISSUE TITLE
Generator dimension mismatch

### ISSUE THREAD

**Issue by ryanbbd** [SayedNadim/Global-and-Local-Attention-Based-Free-Form-Image-Inpainting#10]
Hey, thanks for your job. But I find that, in your model, mask's shape is same as input: 3 * W * H, and in your Generator, the dimension is not right:
`    def forward(self, xin, mask):
        # For indicating the boundaries of images
        ones = torch.ones(xin.size(0), 1, xin.size(2), xin.size(3))
        if self.use_cuda:
            ones = ones.cuda()
            mask = mask.cuda()
        # conv branch
        xnow = torch.cat([xin, ones, mask], dim=1)  #  shape: B * (3+1+3) * H * W
        x = self.conv1(xnow)
################
   class CoarseGenerator(nn.Module):
    def __init__(self, input_dim, cnum, use_cuda=True, device_ids=None):
        super(CoarseGenerator, self).__init__()
        self.use_cuda = use_cuda
        self.device_ids = device_ids

        # 3 x 256 x 256
        self.conv1 = gen_conv(input_dim, cnum, 5, 1, 2)
###############################
netG:
  input_dim: 5
  ngf: 32`
I think the input_dim should be 7, not 5. Could you give me some advice?

**Comment by SayedNadim** [comment]
Hi,
<del>The mask shape is actually [b, 1, h, w]. So, input dim of the generator is 5 [3+1+1]. </del>
I have updated the code to match the mask dimension. I had previously updated the correct mask generation script but maybe uploaded a wrong script while updating other scripts. Thanks for pointing out.
Please let me know if you need any information. 
Cheers!

**Comment by ryanbbd** [comment]
Hey, I've updated your new code. Still meet some problems:
![image](https://user-images.githubusercontent.com/49304722/89653990-88fe1d00-d8fa-11ea-9b9f-600892006c47.png)
I found nothing about the inplace operation....Please take a look for me

**Comment by ryanbbd** [comment]
I find that if I backward d_loss first, then I will meet problem above. But if I backward g_loss first, then nothing will happen

**Comment by SayedNadim** [comment]
Hi, sorry for late reply. Can you please tell me which PyTorch version you are using? The code seems to have an autograd issue for PyTorch version over 1.4.

**Comment by ryanbbd** [comment]
okay， I run the code in two environments: 1.2 and 1.5, they all met this issue. I try to solve it, but failed

**Comment by SayedNadim** [comment]
Hi, can you remove inplace [here](https://github.com/SayedNadim/Global-and-Local-Attention-Based-Free-Form-Image-Inpainting/blob/94351136377af1de11e54810f2ca5bec77f827f8/model/network.py#L308) ?
Let me know if it worked. If not, please share your working code.
